gramps/gui/views/treemodels/peoplemodel.py

```diff
@@ -614,7 +614,7 @@ def get_tree_levels(self):
         return [_('Group As'), _('Name')]
 
     def column_header(self, node):
-        return node.name
+        return node.name if node.name != "" else "[%s]" % _("No surname")
```

Comment:
You could just write `return node.name if node.name else no_surname` and define the `no_surname` variable at the top of the file with `no_surname = config.get("preferences.no-surname-text")`.